pandas/core/groupby/generic.py

```diff
@@ -1760,6 +1763,19 @@ def value_counts(
             )
             result_series = cast(Series, gb.size())
 
+            # GH-46357 Include non-observed categories
+            # of non-grouping columns regardless of `observed`
+            if any(
+                isinstance(grouping.grouping_vector, (Categorical, CategoricalIndex))
+                and not grouping._observed
+                for grouping in groupings
+            ):
+                levels_list = [ping.group_index for ping in groupings]
+                index, _ = MultiIndex.from_product(
+                    levels_list, names=[ping.name for ping in groupings]
+                ).sortlevel()
+                result_series = result_series.reindex(index, fill_value=0)
+
```

Comment:
I also checked to make this change directly in the method `_reindex_output`, so it handles the fact that groupings might have a different `observed` attribute, but it breaks 60 tests.